run_sdk_container

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -eu
+
+cd $(dirname "$0")
+source sdk_lib/sdk_container_common.sh
+
+version="$(get_version_from_versionfile)"
+arch="all"
+checkout="true"
+create_only="false"
+name=""
+
+usage() {
+    echo "  $0 - Start an SDK container of a given SDK release version."
+    echo "       This will create the container if it does not exist, otherwise it will just start it."
+    echo "       If the container is already running then it will exec into the container."
+    echo "  Usage:"
+    echo "  $0 [-v <version>] [-a arch] [-n <name> ] [-c] [-C] [container-command]"
+    echo
+    echo "      container-command - command to be run in the container instead of"
+    echo "                     an interactive shell."
+    echo "      -v <version> - SDK version to use. Defaults to '$version'"
+    echo "                     (from '$sdk_container_common_versionfile')"
+    echo "      -a <amd64|arm64|all> - Target architecture (board support) of the SDK."
+    echo "                     'all' (the default) contains support for both amd64 and arm64."
+    echo "      -n <name>    - Custom name to use for the container."
+    echo "      -c           - Use scripts and ebuild repos as-is, do not check out version tag."
+    echo "      -C           - Create the container if it does not exist, then quit."
+    echo "                     (do not attach, run commands, etc)"
+    echo
+}
+# --
+
+while [ 0 -lt $# ] ; do
+    case "$1" in
+    -h) usage; exit 0;;
+    -v) version="$2"; shift; shift;;
+    -a) arch="$2";    shift; shift;;
+    -n) name="$2";    shift; shift;;
+    -c) checkout="false";    shift;;
+    -C) create_only="true";  shift;;
+    *) break;;
+    esac
+done
+
+if [ -z "$name" ] ; then
+    name="flatcar-sdk-${arch}-${version}"
+fi
+
+lock_version "$version" "$name"
+
+create_versionfile "$version"
+
+if $checkout; then
+    checkout_ebuild_repos "$version"
+fi
+
+stat="$(docker ps --all --no-trunc --filter name="$name" --format '{{.Status}}'\
+        | cut -f1 -d' ')"
+
+if [ -z "$stat" ] ; then
+    yell "Creating a new container '$name'"
+    mkdir -p "__build__/images"
+    exec docker run -ti \
+       -v /dev:/dev \
```

Comment:
On my system, `docker` is aliased with `podman` since most of the `docker` commands / flags are supported by `podman`.

Running `./run_sdk_container`, I got the following:
```
+ exec docker run -ti -v /dev:/dev -v /home/mathieu/github/kinvolk/flatcar-scripts/sdk_container:/mnt/host/source/ -v /home/mathieu/github/kinvolk/flatcar-scripts/__build__/images:/mnt/host/source/src/build -v /home/mathieu/github/kinvolk/flatcar-scripts:/mnt/host/source/src/scripts --privileged --network host -e SDK_USER_ID=1000 -e SDK_GROUP_ID=1001 --name=flatcar-sdk-all-2983.0.0 --hostname flatcar-sdk-all-2983.0.0 ghcr.io/flatcar-linux/flatcar-sdk-all:2983.0.0 /mnt/host/source/sdk_entry.sh
Error: chown `/dev/pts/2`: Operation not permitted: OCI permission denied
```

Since `podman` is rootless, we need to pass the `/dev` as a `--device`:
```diff
     exec docker run -ti \
-       -v /dev:/dev \
+       --device /dev/:/dev/ \
```

Reply:
I think this should even run as root, or? Maybe try `sudo docker`.

Reply:
But maybe it really works to run in a user namespace and the root user gets mapped to your user ID and still can set up loopback devices and create network namespaces, haven't tried this.

Reply:
But the goal here was to use the host's `/dev`, is this required?

Reply:
I can't get `mknod /dev/loop0 b 7 0` work, I think `sudo docker` is required to have the same result in Podman and Docker. For Docker it makes no difference anyway.

By the way, with `sudo podman` `--privileged` implies `-v /dev:/dev` and I think that's also the case for Docker and it can be dropped, or?

Reply:
> But the goal here was to use the host's `/dev`, is this required?

It is for `build_image` (and likely `image_to_vm.sh`) because creating / modifying the OS image uses loopback mounts / devices. Without `--privileged`, access to `/dev/loop[x]` is prohibited, and without `-v /dev:/dev` partition re-scanning after modifications to the block device file does not work in the container, so individual OS image partitions cannot be accessed. A work-around for the latter would be to run udev in the container to update the container device nodes, something I'd rather like to avoid...

Reply:
Ok, `sudo podman -v /dev:/dev --privileged` should be the same as `docker -v /dev:/dev --privileged`.

Reply:
Yes, it seems to work as expected with `sudo` :)